test/integration/nighthawk_test_server.py

```diff
@@ -60,8 +60,14 @@ def waitUntilServerListening(self):
       while tries > 0:
         time.sleep(0.5)
         tries -= 1
-        if sock.connect_ex((self.server_ip, self.server_port)) == 0:
-          return True
+        result = subprocess.run(
+            ["lsof", "-p",
```

Comment:
Why can't the integration test server fail directly vs. us having to try indirectly to measure this? I'm worried about potential races here.